src/utils/Scale.js

```diff
@@ -117,19 +117,40 @@ export function scaleEqual(scaleA, scaleB) {
         _.isEqual(scaleA.range(), scaleB.range());
 }
 
-function indexOfClosestNumberInList(number, list) {
-  return list.reduce((closestI, current, i) => {
-    return Math.abs(current - number) < Math.abs(list[closestI] - number)
-      ? i
-      : closestI;
-  }, 0);
+export function indexOfClosestLeftNumberInList(number, sortedList) {
+  if (sortedList.length <= 1) {
+    return 0;
+  }
+
+  const isDescending = sortedList[0] > sortedList[1];
+  // _.sortedIndex works on ascending lists only
+  // so reverse if working with descending list
+  const listCopy = isDescending ? sortedList.slice().reverse() : sortedList;
```

Comment:
why are you slicing here? shouldn't the sorted list be an array already?

Reply:
lodash's sortedIndex (return index that given number would be inserted in array) only works on arrays sorted ascending.  Reverse mutates the array so we work with a copy. This happens for the Y axis, where the points that d3 maps the labels to are given back in descending order. 
```
_.sortedIndex([1,2,3,4,5], 2.5) =>  2
_.sortedIndex([5,4,3,2,1], 2.5) => 0 //  we want 3 :(
```

Reply:
ahhh, I see. The slice imo is confusing because it's being used for something other than its intended purpose. Can you use `Array.from()` instead? or even `[...sortedList].reverse()`?

Reply:
Interesting, did some perf checking just for fun between destructuring, slice and Array.from and seems like slice wins overall 1st on firefox, safari and very close 2nd on chrome (latest version). I'm usually a bigger proponent for readability but it's definitely possible that someone will use this function with a lot of datapoints (large list). It's also called by the axis', xyplot and label's mouse events (move, enter, leave, click) so it has the potential to be be called many times. I prefer to keep slice() and have the comment above indicate why we're working with a copy and reversing it.

Reply:
ok sounds good :+1: